jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdLoginService.java

```diff
@@ -0,0 +1,143 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.io.IOException;
+import java.security.Principal;
+import javax.security.auth.Subject;
+import javax.servlet.ServletRequest;
+
+import org.eclipse.jetty.security.IdentityService;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.server.UserIdentity;
+import org.eclipse.jetty.util.component.ContainerLifeCycle;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+public class OpenIdLoginService extends ContainerLifeCycle implements LoginService
+{
+    private static final Logger LOG = Log.getLogger(OpenIdLoginService.class);
+
+    private final OpenIdConfiguration _configuration;
+    private final LoginService loginService;
+    private IdentityService identityService;
+    private boolean authenticateNewUsers;
+
+    public OpenIdLoginService(OpenIdConfiguration configuration)
+    {
+        this(configuration, null);
```

Comment:
Is this constructor ever used? Ie is it ever legal to have an OpenIdLoginService without a wrapped LoginService?

Reply:
yes, it is legal to have the `OpenIdLoginService` without the wrapped login service. The wrapped `LoginService` is only added if you need to store roles for users, otherwise it just authenticates the users without any roles